### PR TITLE
chore(web): eslint react hooks part 3

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

The final PR that enables the ESLint React Hook rules.
Also in this PR:
- fixed some svg attribute react errors
- workflow editor scrollbar issue
- product lead banner layout on bigger screens
- simplified the brand form image uploading logic

### Other information (Screenshots)

https://github.com/novuhq/novu/assets/2607232/f81835b1-9db3-46b8-88e1-8690cc90ed3b

Screenshot 2023-08-28 at 12 00 05

https://github.com/novuhq/novu/assets/2607232/bcf6b2d6-423d-4547-bf16-9bdb7372a64d

### CODE REVIEW

File: apps/web/.eslintrc.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ module.exports = {
       },
     ],
     "react-hooks/rules-of-hooks": 'error',
-    "react-hooks/exhaustive-deps": 'off'
+    "react-hooks/exhaustive-deps": 'warn'
```
Comment (LetItRock) [inline apps/web/.eslintrc.js:39]: enabled hook deps warning

File: apps/web/src/components/utils/ProductLead.tsx [hunk 2]
```diff
@@ -13,23 +15,12 @@ export enum ProductLeadVariants {
...
-  ) : (
-    children
-  );
-};
+const WrapperHolder = styled.div<{ variant: ProductLeadVariants }>`
+  display: flex;
+  flex-direction: ${({ variant }) => (variant === ProductLeadVariants.COLUMN ? 'column' : 'row')};
+  justify-content: space-between;
+  gap: 24px;
+`;
```
Comment (LetItRock) [inline apps/web/src/components/utils/ProductLead.tsx:23]: Fixed product lead banner for bigger screens

File: apps/web/src/components/utils/ProductLead.tsx [hunk 3]
```diff
@@ -54,11 +45,16 @@ export const ProductLead = ({
...
   const theme = useMantineTheme();
   const dark = theme.colorScheme === 'dark';
   const isSelfHosted = IS_DOCKER_HOSTED;
-  const segment = useSegment();
+
+  useEffect(() => {
+    segment.track('Banner seen - [Product lead]', {
+      id,
+    });
+  }, [segment, id]);
```
Comment (LetItRock) [inline apps/web/src/components/utils/ProductLead.tsx:57]: the `segment` and `id` doesn't change between re-renders

File: apps/web/src/components/workflow/FlowEditor.tsx [hunk 4]
```diff
@@ -102,6 +102,7 @@ export function FlowEditor({
     setTimeout(() => {
       initializeWorkflowTree();
     }, 0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (LetItRock) [inline apps/web/src/components/workflow/FlowEditor.tsx:105]: some hooks require more careful refactoring, so I decided to leave this comment for now

File: apps/web/src/design-system/icons/actions/PlusFilled.tsx [hunk 5]
```diff
@@ -7,14 +7,14 @@ export function PlusFilled(props: React.ComponentPropsWithoutRef<'svg'>) {
       <path
         d="M7.19995 12H16.8M12 7.20001V16.8"
         stroke="white"
-        stroke-width="2"
-        stroke-linecap="round"
-        stroke-linejoin="round"
+        strokeWidth="2"
+        strokeLinecap="round"
+        strokeLinejoin="round"
```
Comment (LetItRock) [inline apps/web/src/design-system/icons/actions/PlusFilled.tsx:12]: fixed react warning

File: apps/web/src/pages/auth/InvitationPage.tsx [hunk 6]
```diff
@@ -47,13 +47,13 @@ export default function InvitationPage() {
     if (isLoggedInAsInvitedUser) {
       submitToken(tokensRef.current.token as string, tokensRef.current.invitationToken as string, true);
     }
-  }, [isLoggedInAsInvitedUser]);
+  }, [isLoggedInAsInvitedUser, submitToken]);
```
Comment (LetItRock) [inline apps/web/src/pages/auth/InvitationPage.tsx:50]: the `submitToken` is wrapped with the `useCallback` so it doesn't change

File: apps/web/src/pages/auth/components/CreateOrganizationForm.tsx [hunk 7]
```diff
@@ -55,7 +55,7 @@ export function CreateOrganization({}: Props) {
         navigate(ROUTES.HOME);
       }
     }
-  }, []);
+  }, [token, navigate, isFromVercel, startVercelSetup]);
```
Comment (LetItRock) [inline apps/web/src/pages/auth/components/CreateOrganizationForm.tsx:58]: these doesn't change between re-render

File: apps/web/src/pages/auth/components/useAcceptInvite.ts [hunk 8]
```diff
@@ -20,27 +20,30 @@ export function useAcceptInvite() {
...
-
-      navigate(ROUTES.WORKFLOWS);
-    } catch (e: unknown) {
-      errorMessage('Failed to accept an invite.');
-
-      Sentry.captureException(e);
-    }
-  }, []);
+    },
+    [mutateAsync, navigate, queryClient, setToken]
```
Comment (LetItRock) [inline apps/web/src/pages/auth/components/useAcceptInvite.ts:45]: doesn't change between re-render

File: apps/web/src/pages/brand/tabs/BrandingForm.tsx [hunk 9]
```diff
@@ -51,26 +57,17 @@ export function BrandingForm() {
...
 
-  function beforeUpload(files: File[]) {
-    setFile(files[0]);
-  }
-
-  useEffect(() => {
-    if (file) {
-      handleUpload();
-    }
-  }, [file]);
```
Comment (LetItRock) [inline apps/web/src/pages/brand/tabs/BrandingForm.tsx:64]: removed this use effect as it's not needed the action of image upload will be fired on update click

File: apps/web/src/pages/quick-start/components/layout/GetStartedLayout.tsx [hunk 10]
```diff
@@ -23,26 +23,18 @@ export function GetStartedLayout({ children, footer }: IGetStartedLayoutProps) {
...
   useEffect(() => {
-    onRouteChangeUpdateNavigationStore();
+    currentOnboardingStep().set(location.pathname);
   }, [location.pathname]);
 
   useEffect(() => {
-    onStepMountNavigateToCurrentStep();
-  }, []);
-
-  function onStepMountNavigateToCurrentStep() {
```
Comment (LetItRock) [inline apps/web/src/pages/quick-start/components/layout/GetStartedLayout.tsx:33]: no need in these functions that are created on every render

File: apps/web/src/pages/templates/components/email-editor/EmailInboxContent.tsx [hunk 11]
```diff
@@ -25,16 +25,12 @@ export const EmailInboxContent = ({
...
     }
-  }, [layouts]);
-
-  function getDefaultLayout() {
-    const defaultLayout = layouts?.find((layout) => layout.isDefault);
-    setTimeout(() => {
-      setValue(`steps.${index}.template.layoutId`, defaultLayout?._id, { shouldValidate: true });
-    }, 0);
-  }
+  }, [getValues, setValue, layouts, index]);
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/email-editor/EmailInboxContent.tsx:33]: these don't change

**Review by davidsoderberg — APPROVED** [review]
Looks good

**Pull request merged**